src/Fieldset/index.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import newId from '../utils/newId';
+import styles from './Fieldset.scss';
+import ValidationMessage from '../ValidationMessage/index';
+import Variant from '../utils/constants';
+
+const inputProps = {
+  variantIconDescription: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  children: PropTypes.node,
+  className: PropTypes.arrayOf(PropTypes.string),
+  id: PropTypes.string,
+  isValid: PropTypes.bool,
+  legend: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
```

Comment:
Nit: do you mind separating the required and optional `props`? I know I'm OCD, but it makes it a little easier to see at a glance.